MultiLangProgGuide/MultiLangProgGuide.rst

```diff
@@ -1819,6 +1821,920 @@ If the requested :code:`timeout_nsec` exceeds
+  first :code:`int` returned) is not meaningful.
+- Otherwise, return the :code:`len_used` field of the array element
+  specifed by :code:`idx` of the :code:`C.ydb_buffer_t` array referenced
+  by :code:`cbuftary` with a return code of :code:`C.YDB_OK`.
+
+Go BufferTArray GetUsed()
+-------------------------
+
+.. code-block:: go
+
+    GetUsed() int
+
+Returns the value of the :code:`elemsUsed` field.
+
+Go BufferTArray GetValBAry()
+----------------------------
+
+.. code-block:: go
+
+    GetValBAry(idx int) *[]byte, int
+
+- If :code:`idx` is greater than the :code:`elemsAlloc` of the
+  :code:`BufferTArray` structure, return with a return code of
+  :code:`C.YDB_ERR_INSUFFSUBS`. In this case, the return value (the
+  first :code:`int` returned) is not meaningful.
+- In the event the :code:`len_alloc` field of the
+  :code:`C.ydb_buffer_t` structure is less than its :code:`len_used`
+  field (as a consequence of a previous :code:`C.YDB_ERR_INVSTRLEN`
+  error), return with a return code of :code:`C.YDB_ERR_INVSTRLEN`. In
+  this case, the return value (the first :code:`int` returned) is not
+  meaningful.
+- Otherwise, return the value of the buffer referenced by the
+  :code:`C.ydb_buffer_t` structure at the element specified by
+  :code:`idx` and referenced by :code:`cbuftary` as a byte array with
+  a return code of :code:`C.YDB_OK`.
+
+Go BufferTArray GetValStr()
+---------------------------
+
+.. code-block:: go
+
+    GetValStr(idx int) *string, int
+
+- If :code:`idx` is greater than the :code:`elemsAlloc` of the
+  :code:`BufferTArray` structure, return with a return code of
+  :code:`C.YDB_ERR_INSUFFSUBS`. In this case, the return value (the
+  first :code:`int` returned) is not meaningful.
+- In the event the :code:`len_alloc` field of the
+  :code:`C.ydb_buffer_t` structure is less than its :code:`len_used`
+  field (as a consequence of a previous :code:`C.YDB_ERR_INVSTRLEN`
+  error), return with a return code of :code:`C.YDB_ERR_INVSTRLEN`. In
+  this case, the return value (the first :code:`int` returned) is not
+  meaningful.
+- Otherwise, return the value of the buffer referenced by the
+  :code:`C.ydb_buffer_t` structure at the element specified by
+  :code:`idx` and referenced by :code:`cbuftary` as a string with
+  a return code of :code:`C.YDB_OK`.
+
+Go BufferTArray SetUsed()
+-------------------------
+
+.. code-block:: go
+
+    SetUsed(newidx int) int
+
+- If :code:`idx` is greater than the :code:`elemsAlloc` of the
+  :code:`BufferTArray` structure, return with a return code of
+  :code:`C.YDB_ERR_INSUFFSUBS`.
+- Otherwise, set:code:`elemsUsed` to :code:`newidx` and return
+  :code:`C.YDB_OK`
+
+Note that even if :code:`idx` is not greater than the value of
+:code:`elemsAlloc`, setting :code:`elemsUsed` to a value greater than
+the number of valid values in the array will likely lead to
+hard-to-debug errors. The intended use of this function is when an
```

Comment:
This sounds as if it is **only** used when one wants to set a subset of an existing array. There are two choices here:

1. The function is always required to set the number of elements set in the array.
2. The Set* methods keep track of a high-water mark of indexes that have been set and use that unless this function resets the used count - the purpose of to either use a subset of an existing array or to set the used count to zero so the array can be "repopulated".

Personally, I like option #1 better as it sets an expectation with the user they would likely expect anyway and eliminates complexity at a very minor cost.